### PR TITLE
Adjust user group indexes.

### PR DESCRIPTION
**Pull request by agorina**
Why? I am reviewing `slog_log` in both QA and UAT and modifying some indexes based on it. There are more to come.

### CODE REVIEW

File: reporting/sql/V1_2_0_11__user_groups-index.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+use ${schemaName};
+
+ALTER TABLE user_student_group
+ DROP FOREIGN KEY fk__user_student_group__student_group,
+ DROP INDEX idx__user_student_group;
```
Comment (agorina) [inline reporting/sql/V1_2_0_11__user_groups-index.sql]: Before:  
``` 
UNIQUE INDEX idx__user_student_group (student_group_id, user_login),
CONSTRAINT fk__user_student_group__student_group FOREIGN KEY (student_group_id) REFERENCES student_group(id)```

File: reporting/sql/V1_2_0_11__user_groups-index.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+ALTER TABLE user_student_group
+  -- reverse the order of columns in the index
+  ADD UNIQUE KEY idx__user_student_group(user_login, student_group_id),
+  -- add index to support FK
+  ADD INDEX idx__user_student_group__student_group(student_group_id),
+  ADD CONSTRAINT fk__user_student_group__student_group FOREIGN KEY (student_group_id) REFERENCES student_group(id);
```
Comment (agorina) [inline reporting/sql/V1_2_0_11__user_groups-index.sql]: The logged query was:
```
select
  sg.id,
  sg.name,
  sc.name as school_name,
  sc.id as school_id,
  s.code as subject_code
from user_student_group usg
  join student_group sg on sg.id = usg.student_group_id
  join school sc on sc.id = sg.school_id
  left join subject s on s.id = sg.subject_id
where user_login='nacostaDP2@example.com';
```
There are 1,048,782 use groups in QA.

Before:
before

After:
after

Comment (malaffoon) [reply reporting/sql/V1_2_0_11__user_groups-index.sql]: nice

**Comment by agorina** [comment]
Merging but will address comments if needed.

**Pull request merged**